pkg/user/apiserver/registry/user/etcd/etcd.go

```diff
@@ -101,10 +100,16 @@ func (r *REST) Get(ctx context.Context, name string, options *metav1.GetOptions)
 
 	// do not bother looking up users that cannot be persisted
 	// make sure we return a status error otherwise the API server will complain
-	if reasons := apivalidation.ValidateUserName(name, false); len(reasons) != 0 {
-		err := field.Invalid(field.NewPath("metadata", "name"), name, strings.Join(reasons, ", "))
-		return nil, kerrs.NewInvalid(usergroup.Kind("User"), name, field.ErrorList{err})
+	if err := validateUsername(name); err != nil {
+		return nil, err
 	}
 
 	return r.Store.Get(ctx, name, options)
 }
+
+func validateUsername(name string) *kerrs.StatusError {
+	if errorList := validation.ValidateUserNameField(name, field.NewPath("metadata", "name")); len(errorList) != 0 {
+		return kerrs.NewInvalid(usergroup.Kind("User"), name, errorList)
+	}
+	return nil
+}
```

Comment:
Inline it back, please. I don't see a reason to keep this outside.

Reply:
I've extracted the validation part to make it unit-testable. Otherwise, I'd need to bootstrap the whole REST server.

Reply:
Let me get this back. I'll inline the code and move the tests.

Reply:
Updated, thanks!